lib/devices/ios/ios-log.js

```diff
@@ -44,91 +40,112 @@ IosLog.prototype.startCapture = function (cb) {
+
+        // deviceconsole retrieves many old device log lines that came before it was
+        // started so filter those out until we encounter new log lines.
+        this.proc = spawn("deviceconsole", ["-u", this.udid], {env: spawnEnv});
       }
-      var errOnlyCb = function (cb) { return function (err) { cb(err); }; }; // makes waterfall safer
-      var touchLog = function (cb) {
-        async.series([
-          function (cb) { mkdirp(logsPath, errOnlyCb(cb)); },
-          function (cb) { touch(path.resolve(logsPath, "system.log"), errOnlyCb(cb)); },
-          function (cb) {
-            require('fs').appendFile(
-              path.resolve(logsPath, "system.log"),
-              'A new simulator is about to start!',
-              errOnlyCb(cb));
+
+      this.finishStartingLogCapture(cb);
+    }.bind(this));
+
+  } else {
+    xcode.getVersion(function (err, xcodeVersion) {
+      if (err) return cb(err);
+
+      var ver = parseInt(xcodeVersion.split(".")[0], 10);
+      if (ver >= 5) {
+        var logsPath;
+        if (ver >= 6) {
+          logger.debug("Starting iOS 8.* simulator log capture");
+          if (_.isUndefined(this.simUdid)) {
+            return cb(new Error("iOS8 log capture requires a sim udid"));
           }
-        ], function (err) { cb(err); });
-      };
-      async.waterfall([
-        function (cb) {
-          if (logsPath.indexOf('*') < 0) {
-            // there is no doubt on system.log location
-            touchLog(cb);
-          } else cb();
-        }, function (cb) {
-          glob(path.resolve(logsPath, "system.log"), function (err, files) {
-            if (err || files.length < 1) {
-              logger.error("Could not start log capture because no iOS " +
-                "simulator logs could be found at " + logsPath + "/system.log" +
-                "Logging will not be functional for this run");
-              return cb(new Error('Could not start log capture'));
-            }
-            cb(null, files);
-          });
-        }, function (files, cb) {
-          var lastModifiedLogPath = files[0]
-            , lastModifiedLogTime = fs.statSync(files[0]).mtime;
-          _.each(files, function (file) {
-            var mtime = fs.statSync(file).mtime;
-            if (mtime > lastModifiedLogTime) {
-              lastModifiedLogPath = file;
-              lastModifiedLogTime = mtime;
+          logsPath = path.resolve(process.env.HOME, "Library", "Logs",
+                                  "CoreSimulator", this.simUdid);
+        } else {
+          logger.debug("Starting iOS 7.* simulator log capture");
+          logsPath = path.resolve(process.env.HOME, "Library", "Logs",
+                                  "iOS Simulator", "7.*");
+        }
+        var errOnlyCb = function (cb) { return function (err) { cb(err); }; }; // makes waterfall safer
+        var touchLog = function (cb) {
+          async.series([
```

Comment:
not sure we need `errOnlyCb` in `async.series`. I thought the default behavior for `series` was already that. It's `waterfall` we need to worry about, right?